repo-scripts/size-analysis/analysis-helper.ts

```diff
@@ -225,52 +226,117 @@ export function extractDeclarations(
           });
         }
       });
+    } else if (ts.isImportDeclaration(node) && node.importClause) {
+      const symbol = checker.getSymbolAtLocation(node.moduleSpecifier);
+      if (symbol && symbol.valueDeclaration) {
+        const importFilePath = symbol.valueDeclaration.getSourceFile().fileName;
+        // import { a, b } from '@firebase/dummy-exp'
+        if (
+          node.importClause.namedBindings &&
+          ts.isNamedImports(node.importClause.namedBindings)
+        ) {
+          node.importClause.namedBindings.elements.forEach(each => {
+            const symbolName: string = each.name.getText(sourceFile);
+            symbolLocation.set(symbolName, importFilePath);
+          });
+          // import * as fs from 'fs'
+        } else if (
+          node.importClause.namedBindings &&
+          ts.isNamespaceImport(node.importClause.namedBindings)
+        ) {
+          const symbolName: string = node.importClause.namedBindings.name.getText(
+            sourceFile
+          );
+          symbolLocation.set(symbolName, importFilePath);
+          // import a from '@firebase/dummy-exp'
+        } else if (
+          node.importClause.name &&
+          ts.isIdentifier(node.importClause.name)
+        ) {
+          const symbolName: string = node.importClause.name.getText(sourceFile);
+          symbolLocation.set(symbolName, importFilePath);
+        }
+      }
     }
     // re-exports handler: handles cases like :
+    // export {LogLevel};
     // export * from '..';
     // export {foo, bar} from '..';
     // export {foo as foo1, bar} from '...';
     else if (ts.isExportDeclaration(node)) {
-      if (node.moduleSpecifier && ts.isStringLiteral(node.moduleSpecifier)) {
-        const symbol = checker.getSymbolAtLocation(node.moduleSpecifier);
-        const reExportFullPath = symbol.valueDeclaration.getSourceFile()
-          .fileName;
-
-        // first step: always retrieve all exported symbols from the source location of the re-export.
-        const reExportsMember = extractDeclarations(reExportFullPath);
-
-        // named exports: eg: export {foo, bar} from '...'; and export {foo as foo1, bar} from '...';
+      if (node.moduleSpecifier) {
```

Comment:
This block has grown too big and too complex with nested if else. It's very hard for me to review. Can you extract it into a few smaller functions each of which handles a single use case (e.g. one for directly export, one for reexport, one for import then export,  etc). Also please add comments to help people understand.